Aplib.Extensions/Actions/PathfinderAction.cs

```diff
@@ -0,0 +1,43 @@
+using Aplib.Core;
+using Aplib.Core.Belief.BeliefSets;
+using System;
+
+namespace Aplib.Extensions.Actions
+{
+    public class PathfinderAction<TBeliefset, TLocation> : Core.Intent.Actions.Action<TBeliefset>
+        where TBeliefset : IBeliefSet
+    {
+        public PathfinderAction(IMetadata metadata,
+            IPathfinder<TLocation> pathfinder,
+            Func<TBeliefset, TLocation> getCurrentLocation,
+            Func<TBeliefset, TLocation> getEndLocation,
+            Action<TBeliefset, TLocation> effect)
```

Comment:
Something like this?
```suggestion
            Action<TBeliefset, TLocation> moveToNextLocation)
```

Reply:
Not really, it is still the effect of the action, you can do with it what you want.